coderd/users.go

```diff
@@ -818,7 +818,16 @@ func generateAPIKeyIDSecret() (id string, secret string, err error) {
 	return id, secret, nil
 }
 
-func (api *API) createAPIKey(rw http.ResponseWriter, r *http.Request, params database.InsertAPIKeyParams) (string, bool) {
+type createAPIKeyParams struct {
+	UserID    uuid.UUID
+	LoginType database.LoginType
+
+	// Optional.
+	ExpiresAt       time.Time
+	LifetimeSeconds int64
+}
```

Comment:
Good abstraction :+1: Aaaaand it's way cleaner now that we don't pass tokens to each API key.